### PR TITLE
Cat service

### PR DESCRIPTION
**Pull request by Espositosole**
This pull request introduces a new `CatsService` module, along with associated models and comprehensive test coverage for managing cats in the system. It also includes updates to existing adopter-related tests to improve structure and maintain consistency. Below are the key changes grouped by theme.

### New Features for Cats Management:
* Added `CatsModel` and `UpdateCatsModel` interfaces to define the structure of cat-related data (`CatsModel.ts`, `UpdateCatsModel.ts`). [[1]](diffhunk://#diff-63a2840246465f8be9ccbb4ae0b805af0d698e019ea57b7779224ba1aa177ef0R1-R13) [[2]](diffhunk://#diff-da57960532519105f6f078829042e4b3c1a140e07e20f87e90951222a37e2395R1-R4)
* Implemented `CatsService` with methods for CRUD operations and filtering cats by parameters such as adoption status and temperament (`CatsService.ts`).

### Test Suite Additions for Cats:
* Created tests for adding, deleting, and retrieving cats, including edge cases like invalid input and non-existent IDs (`addCat.spec.ts`, `deleteCat.spec.ts`, `getCat.spec.ts`). [[1]](diffhunk://#diff-65c8cba68dfe0151508425d71857b2af3cec3c10feb9d92340633dd332e1ecb5R1-R216) [[2]](diffhunk://#diff-8aa053424cd818b44c3ec825a20461c4c95f993835ab834435180ac5b9abd416R1-R31) [[3]](diffhunk://#diff-920a9c5b9c52b1fb58753119f5def72fa213bfc2dfc343487462154a792ab795R1-R84)

### Refactoring of Adopter Tests:
* Renamed adopter test files to follow a consistent naming convention and updated import paths for better organization (`AdopterService/AddAdopter.spec.ts`, `AdopterService/DeleteAdopter.spec.ts`, `AdopterService/GetAdopterById.spec.ts`, `AdopterService/GetAdopters.spec.ts`). [[1]](diffhunk://#diff-aff9daa5571be477932d995cedd8e33cb782e469322dd9cc83e8f05067034a44L1-R5) [[2]](diffhunk://#diff-0c1f3d489e51e1dda2038bdd06262ef8ec256a63bdba00d9001930990daf0a2bL1-R3) [[3]](diffhunk://#diff-e213316d471936e078d11b77aeac88ddc377f50b3084ad0ac0f555cd26865fbdL1-R4) [[4]](diffhunk://#diff-1fe75ee2aa15dbb0861a69ea75a5c74064aa9c3caed287d1f56df06f858a38e8L1-R2)
* Minor formatting adjustments in test cases for readability and consistency (e.g., breaking long lines in `AddAdopter.spec.ts`). [[1]](diffhunk://#diff-aff9daa5571be477932d995cedd8e33cb782e469322dd9cc83e8f05067034a44L31-R32) [[2]](diffhunk://#diff-aff9daa5571be477932d995cedd8e33cb782e469322dd9cc83e8f05067034a44L47-R49) [[3]](diffhunk://#diff-aff9daa5571be477932d995cedd8e33cb782e469322dd9cc83e8f05067034a44L68-R73) [[4]](diffhunk://#diff-aff9daa5571be477932d995cedd8e33cb782e469322dd9cc83e8f05067034a44L106-R112)

### Bug Tracking:
* Marked specific test cases as skipped due to known bugs, with references to GitHub issues for tracking (e.g., invalid `dateJoined` and `staffInCharge` bugs in `addCat.spec.ts`).

These changes collectively enhance the functionality and maintainability of the codebase by introducing robust support for managing cats and improving test structure for adopters.

### CODE REVIEW

File: framework/src/models/request/CatsModel.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    name: string,
+    age: number,
+    breed: string,
+    dateJoined: string,
+    vaccinated: boolean,
+    temperament: string[],
+    staffInCharge: string,
+    isAdopted: boolean,
+    adopterId?: number,
+
```
Comment (damianpereira86) [inline framework/src/models/request/CatsModel.ts]: Remove empty line

**Review by damianpereira86 — APPROVED** [review]
